lib/kwtool/crawler/keywords.ex

```diff
@@ -36,16 +31,54 @@ defmodule Kwtool.Crawler.Keywords do
     |> Repo.paginate(params)
   end
 
-  def get_keyword_by_user(%User{} = user, keyword_id) do
+  def find_keyword_by_user(%User{} = user, keyword_id) do
     Keyword
     |> where([k], k.user_id == ^user.id)
     |> where([k], k.id == ^keyword_id)
+    |> preload(:keyword_results)
     |> Repo.one()
   end
 
+  def find_by_id!(keyword_id) do
+    Keyword
+    |> where(id: ^keyword_id)
+    |> Repo.one!()
+  end
+
+  def create_keyword_result(%Keyword{id: keyword_id}, crawler_result_attrs) do
+    crawler_result_attrs
+    |> Map.put(:keyword_id, keyword_id)
+    |> KeywordResult.create_changeset()
+    |> Repo.insert()
+  end
+
+  def mark_as_finished(%Keyword{} = keyword) do
+    keyword
+    |> Keyword.finished_changeset()
+    |> Repo.update()
+  end
+
+  defp process_keyword_list(keyword_list, user_id) do
+    Enum.each(keyword_list, fn keyword ->
+      {:ok, %Keyword{id: keyword_id}} =
```

Comment:
Should this handle when fail to create keyword?

Reply:
Updated in https://github.com/byhbt/kwtool/pull/44/commits/8397dc85de9d2e75b4014c26d781f0d5d5b03763